contracts/APPCoin.sol

```diff
@@ -59,7 +60,21 @@ contract APPCoin is Initializable, ERC777Upgradeable, PausableUpgradeable, Ownab
             delete frozenMap[acc];
         }
     }
+    function charge(address account, uint256 amount, bytes memory data) public onlyOwner {
+        _burn(account, amount, "", data);
+        if (frozenMap[account] > 1) {
+            // refund
+            uint256 appCoinLeft = balanceOf(account);
+            _burn(account, appCoinLeft, "", "refund");
+            IERC777Upgradeable(apiCoin).send(account, appCoinLeft, "refund");
+            delete frozenMap[account];
+            emit Withdraw(account, appCoinLeft);
+        }
```

Comment:
Does this make any loophole to bypass the time lock?

Reply:
Refunding after charging is normal logic, as discussed.